devices/ble_hci/common-hal/_bleio/att.c

```diff
@@ -1713,77 +1662,3 @@ void att_process_data(uint16_t conn_handle, uint8_t dlen, uint8_t data[]) {
             break;
     }
 }
-
-// FIX Do we need all of these?
-void check_att_err(uint8_t err) {
-    const compressed_string_t *msg = NULL;
-    switch (err) {
-        case 0:
-            return;
-        case BT_ATT_ERR_INVALID_HANDLE:
-            msg = translate("Invalid handle");
-            break;
-        case BT_ATT_ERR_READ_NOT_PERMITTED:
-            msg = translate("Read not permitted");
-            break;
-        case BT_ATT_ERR_WRITE_NOT_PERMITTED:
-            msg = translate("Write not permitted");
-            break;
-        case BT_ATT_ERR_INVALID_PDU:
-            msg = translate("Invalid PDU");
-            break;
-        case BT_ATT_ERR_NOT_SUPPORTED:
-            msg = translate("Not supported");
-            break;
-        case BT_ATT_ERR_INVALID_OFFSET:
-            msg = translate("Invalid offset");
-            break;
-        case BT_ATT_ERR_PREPARE_QUEUE_FULL:
-            msg = translate("Prepare queue full");
-            break;
-        case BT_ATT_ERR_ATTRIBUTE_NOT_FOUND:
-            msg = translate("Attribute not found");
-            break;
-        case BT_ATT_ERR_ATTRIBUTE_NOT_LONG:
-            msg = translate("Attribute not long");
-            break;
-        case BT_ATT_ERR_ENCRYPTION_KEY_SIZE:
-            msg = translate("Encryption key size");
-            break;
-        case BT_ATT_ERR_INVALID_ATTRIBUTE_LEN:
-            msg = translate("Invalid attribute length");
-            break;
-        case BT_ATT_ERR_UNLIKELY:
-            msg = translate("Unlikely");
-            break;
-        case BT_ATT_ERR_UNSUPPORTED_GROUP_TYPE:
-            msg = translate("Unsupported group type");
-            break;
-        case BT_ATT_ERR_INSUFFICIENT_RESOURCES:
-            msg = translate("Insufficient resources");
-            break;
-        case BT_ATT_ERR_DB_OUT_OF_SYNC:
-            msg = translate("DB out of sync");
-            break;
-        case BT_ATT_ERR_VALUE_NOT_ALLOWED:
-            msg = translate("Value not allowed");
-            break;
-    }
-    if (msg) {
-        mp_raise_bleio_BluetoothError(msg);
-    }
-
-    switch (err) {
-        case BT_ATT_ERR_AUTHENTICATION:
-            msg = translate("Insufficient authentication");
-            break;
-        case BT_ATT_ERR_INSUFFICIENT_ENCRYPTION:
-            msg = translate("Insufficient encryption");
-            break;
-    }
-    if (msg) {
-        mp_raise_bleio_SecurityError(msg);
-    }
-
-    mp_raise_bleio_BluetoothError(translate("Unknown ATT error: 0x%02x"), err);
-}
```

Comment:
Please don't delete this; the _bleio HCI implementation is incomplete and this may be needed later. Let's just comment it out for now.

Reply:
I reverted these changes.  Instead, I marked the functions as static but deactivated the diagnostic for unused functions. This should be the most effective way to ensure the functions' code doesn't go into binaries, while leaving the code available for the future when it's needed.